pgmpy/continuous/discretize/base.py

```diff
@@ -0,0 +1,45 @@
+import numpy as np
+
+from pgmpy.factors import ContinuousFactor
+
+
+class BaseDiscretizer(object):
+    """
+    Base class for the discretizer classes in pgmpy. The discretizer
+    classes are used to discretize a continuous random variable
+    distribution into discrete probability masses.
+
+    Parameters
+    ----------
+    factor: ContinuousFactor
+        the continuous factor representing the distribution
+        to be discretized.
+    frm, to: float
+        the range over which the function will be discretized.
+    step: float
+        the discretization step (or span, or lag)
+
+    """
+
+    def __init__(self, factor, frm, to, step):
```

Comment:
@ankurankan Also I was trying to keep the API similar to R {actuar} discretize methods. Should I just add step as a separate parameter for these methods?